### PR TITLE
[Merged by Bors] - refactor: review normed group structure on `ContinuousLinearMap`

### PR DESCRIPTION
**Pull request by urkud**
Move parts of the proof of "two uniformities are equal" up while generalizing it.
Also add a `ContinuousConstSMul` instance that needs less assumptions than `ContinuousSMul`.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Analysis/NormedSpace/OperatorNorm.lean [hunk 1]
```diff
@@ -1888,9 +1830,12 @@ def smulRightL : (E →L[𝕜] 𝕜) →L[𝕜] Fₗ →L[𝕜] E →L[𝕜] F
...
           LinearMap.smul_apply, RingHom.id_apply, Pi.smul_apply] }
-    1 fun c x => by
-      simp only [coe_smulRightₗ, one_mul, norm_smulRight_apply, LinearMap.coe_mk, AddHom.coe_mk,
-        le_refl]
+    1 fun c x => (norm_smulRight_apply c x).le.trans_eq <| by simp
+    -- was
+    -- simp only [coe_smulRightₗ, one_mul, norm_smulRight_apply, LinearMap.coe_mk, AddHom.coe_mk,
+    --   le_refl]
+    -- after `=>` above
+    -- Now it fails to use `AddHom.coe_mk`. WHY?
```
Comment (urkud) [inline Mathlib/Analysis/NormedSpace/OperatorNorm.lean:1838]: @digama0 Could you please have a look?

**Comment by urkud** [comment]
bench!

**Comment by hrmacbeth** [comment]
bors d+

Nice!  If there are no performance issues, please merge.

**Comment by hrmacbeth** [comment]
!bench

**Pull request closed without merging**